### PR TITLE
github actions: add package list generation

### LINKED ISSUE #146: push-button generation of package version list
Add a Github Action to this repo which:

- Clones the project website repo (this repo)
- Clones the Bottlerocket OS repo at a specified tag (patch release)
- Runs the package version inventory script (`scripts/software-versions-inventory/software-versions-inventory.sh`) and generates a new package version list page under `content/en/os/MINOR_RELEASE/version-information/packages/PATCH_RELEASE/`
- Opens a PR for that new package version list page

### PR DESCRIPTION
**Pull request by mchaker**
**Issue number:**

Closes #146 

**Description of changes:**

Adds a Github Action to this repo which:

- Clones this repo
- Clones the Bottlerocket OS repo at a specified tag (patch release)
- Generates the package version list for that patch release
- Makes a new branch on this repo for that new package version list page
- Opens a PR against this repo for the new package version list page

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: .github/workflows/generate-package-list.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          repository: bottlerocket-os/bottlerocket
+          path: bottlerocket
+          ref: v${{ inputs.patchRelease }}
+      - name: Remove old package list
+        run: |
+          rm -rf bottlerocket-project-website/content/en/os/$MINOR_RELEASE/version-information/packages/$PATCH_RELEASE/
+          find bottlerocket-project-website/content/en/os/$MINOR_RELEASE/version-information/packages/
+        env:
+          MINOR_RELEASE: ${{ inputs.minorRelease }}
+          PATCH_RELEASE: ${{ inputs.patchRelease }}
```
Comment (mchaker) [inline .github/workflows/generate-package-list.yml:39]: Unsure if this is necessary?

Comment (stockholmux) [reply .github/workflows/generate-package-list.yml:39]: `PATCH_RELEASE: ${{ inputs.patchRelease }}`?

File: .github/workflows/generate-package-list.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Generate package version list
+        run: |
+          bash bottlerocket-project-website/scripts/software-versions-inventory/software-versions-inventory.sh bottlerocket/ > bottlerocket-project-website/content/en/os/$MINOR_RELEASE/version-information/packages/$PATCH_RELEASE/index.markdown
+        env:
+          MINOR_RELEASE: ${{ inputs.minorRelease }}
+          PATCH_RELEASE: ${{ inputs.patchRelease }}
+      - name: Open PR for changes
+        uses: peter-evans/create-pull-request@v5
```
Comment (stockholmux) [inline .github/workflows/generate-package-list.yml:53]: MIT licensed 👍

File: .github/workflows/generate-package-list.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        with:
+          path: bottlerocket-project-website
+          commit-message: 'packages: add package versions for v${{ inputs.patchRelease }}'
+          branch: add-package-versions-${{ inputs.patchRelease }}
+          title: 'packages: add package versions for v${{ inputs.patchRelease }}'
+          body: |
+            **Issue number:**
```
Comment (stockholmux) [inline .github/workflows/generate-package-list.yml:60]: This seems to be a copy/paste of the PR template. I don't suppose there is a way to keep this synchronized with the actual PR template? (e.g. if I change the PR template, this workflow also changes)

**Review by stockholmux — COMMENTED** [review]
A few comments, but it looks good other wise.

File: .github/workflows/generate-package-list.yml (hunk 3 above)
Comment (mchaker) [reply .github/workflows/generate-package-list.yml:60]: There doesn't seem to be a way to add parameters/input values when using a template file for the PR body.

File: .github/workflows/generate-package-list.yml (hunk 1 above)
Comment (mchaker) [reply .github/workflows/generate-package-list.yml:39]: Removing this step

File: .github/workflows/generate-package-list.yml (hunk 3 above)
Comment (stockholmux) [reply .github/workflows/generate-package-list.yml:60]: OK. It's a nice to have. Could you add a comment to `/.github/PULL_REQUEST_TEMPLATE` to warn anyone looking to change that file they'll also need to update this workflow?

File: .github/workflows/generate-package-list.yml (hunk 1 above)
Comment (stockholmux) [reply .github/workflows/generate-package-list.yml:39]: ahh. I get it. Yeah, let's retain the removal

File: .github/workflows/generate-package-list.yml (hunk 3 above)
Comment (mchaker) [reply .github/workflows/generate-package-list.yml:60]: Yes - adding a comment to that template. Thanks!

**Pull request merged**